src/CodeGen_LLVM.cpp

```diff
@@ -1439,6 +1444,11 @@ void CodeGen_LLVM::visit(const Cast *op) {
         debug(4) << "Emulating cast from " << src << " to " << dst << "\n";
         if ((src.is_float() && src.bits() < 32) ||
             (dst.is_float() && dst.bits() < 32)) {
+            string warn_msg =
+                "(b)float16 type operation is emulated, which is likely to slow down the performance. "
+                "If your target supports native (b)float16 operations, "
+                "it could be improved by adding Target feature to enable it.\n";
+            onetime_warnings.try_emplace(__LINE__, warn_msg);
```

Comment:
Why use __LINE__ here? Presumably it will always be the same value (i.e., 1451 at this point in code history)

Reply:
It was the mechanism for making it one-time, as you think, but I found it would be a problem if the same line number in different files add different warnings. Now it is replaced with enum.